vault/logical_system.go

```diff
@@ -3048,7 +3048,8 @@ func (b *SystemBackend) pathInternalUIMountRead(ctx context.Context, req *logica
 	// Load the ACL policies so we can walk the prefix for this mount
 	acl, te, entity, _, err := b.Core.fetchACLTokenEntryAndEntity(ctx, req)
 	if err != nil {
-		return nil, err
+		b.logger.Warn("permission denied as the token is missing or invalid")
```

Comment:
I think this log message is misleading because we could have got any number of other errors that aren't related to missing or invalid tokens.  Actually I'd rather not have a log message here at all, because we usually only write to system log when servicing requests and something has gone wrong internally, not when the request itself is problematic.

Similarly, you're now always returning a 400, and sometimes we might get an error that actually warrants a 500.

How about instead you check explicitly here for missing client token (like at the start of fetchACLTokenEntryAndEntity) and only return ErrPermissionDenied in that case?  Doesn't handle every permission denied scenario, but at least it's never incorrect.

Reply:
That sounds like the right approach. `fetchACLTokenEntryAndEntity` is called from several places, I'm going to look into changing it to return a pre-declared missingToken error of some kind, and then decide which places need to inspect the error